src/status_im2/contexts/wallet/events.cljs

```diff
@@ -0,0 +1,40 @@
+(ns status-im2.contexts.wallet.events
+  (:require [utils.re-frame :as rf]
+            [taoensso.timbre :as log]
+            [utils.security.core :as security]
+            [native-module.core :as native-module]))
+
+(rf/reg-event-fx 
+:wallet-2/create-derived-addresses
+                 (fn [{:keys [db]} [password {:keys [path]} on-success]]
+                   (let [{:keys [wallet-root-address]} (:profile/profile db)
+                         sha3-pwd   (native-module/sha3 (str (security/safe-unmask-data password)))]
+                     {:json-rpc/call [{:method     "wallet_getDerivedAddresses"
+                                       :params     [sha3-pwd (clj->js wallet-root-address) (clj->js [path])]
+                                       :on-success on-success
+                                       :on-error   #(log/info "failed to derive address " %)}]})))
+(rf/reg-event-fx :wallet-2/add-account
+                 (fn [{:keys [db]} [password {:keys [emoji account-name color]} {:keys [public-key address  path]}]]
+                   (let [key-uid   (get-in db [:profile/profile :key-uid])
+                         sha3-pwd   (native-module/sha3  (security/safe-unmask-data password))
+                         account-config (clj->js {:key-uid key-uid
+                                                  :wallet false
+                                                  :chat false
+                                                  :type :generated
+                                                  :name account-name
+                                                  :emoji emoji
+                                                  :path path
+                                                  :address  address
+                                                  :public-key public-key
+                                                  :colorID color})]
+                     {:json-rpc/call [{:method     "accounts_addAccount"
```

Comment:
To follow the new guidelines on events, the code is slightly different, using `:fx`. `:fx` is the only effect we need to describe all other effects (ordered) in the event handler.

```clojure
(rf/reg-event-fx :wallet-2/add-account
 (fn [{:keys [db]} [password {:keys [emoji account-name color]} {:keys [public-key address path]}]]
   (let [key-uid        (get-in db [:profile/profile :key-uid])
         sha3-pwd       (native-module/sha3 (security/safe-unmask-data password))
         account-config (clj->js {:key-uid    key-uid
                                  :wallet     false
                                  :chat       false
                                  :type       :generated
                                  :name       account-name
                                  :emoji      emoji
                                  :path       path
                                  :address    address
                                  :public-key public-key
                                  :colorID    color})]
     {:fx [[:json-rpc/call
            [{:method     "accounts_addAccount"
              :params     [sha3-pwd account-config]
              :on-success #(rf/dispatch [:navigate-to :wallet-accounts])
              :on-error   #(log/info "failed to create account " %)}]]]})))
```

Reply:
@J-Son89  @ilmotta I'm not sure about this.
If the order is not important (like right now that th effect returned is just one) I think we could describe the effect at the root of the map returned.

I mean, it's less verbose and the result is the same. We could use the `:fx` syntax:
```clojure
{:fx [ [:effect-1 params-1] [:effect-2 params-2] ,,,]
 ,,,}
```
when we need to describe many effects **AND** the order is important.
WDYT?

Reply:
But why have two ways to do the same thing @ulisesmac? The old API only exists for backwards compatibility. The recommendation from re-frame is pretty solid and we added it to our guidelines too. All the docs in re-frame refer to the `:fx` effect. If we stick to vanilla re-frame we will be following the official docs people read.

> While event handlers can return a map of arbitrary effects, it is now recommended that they only return a map containing two standard keys :db and :fx.
>
> -- https://github.com/day8/re-frame/blob/39adca93673f334dc751ee2d99d340b51a9cc6db/docs/FAQs/BestPractice.md#use-the-fx-effect

This `now` the re-frame doc refers to is already 3 years old btw.

> when we need to describe many effects AND the order is important.

This puts an additional burden on devs to think about which syntax to use. Order in effects is 99% of the time desirable. Even when we don't need, some time in another PR, a developer might need, and they would need to refactor the code to use the `:fx` effect. Obviously devs will probably forget about this, so it's better to just stick to the best practice.

Reply:
Got it, makes sense to have uniform code, also stick to best practices, as far as I remembered, returning only `:fx` was just a suggestion, not the best practice, now I see this is different.

(btw, I'm still not happy about how re-frame's API has evolved)

Reply:
> (btw, I'm still not happy about how re-frame's API has evolved)

I started to appreciate the API more after reading this long issue and all the insightful comments. Check out issue 639 in https://github.com/day8/re-frame/issues @ulisesmac

Reply:
OMG, a long issue. I will take a look :+1:  thanks!